lua/websearcher.lua

```diff
@@ -1,60 +1,259 @@
+            command()
+        else
+            os.execute(command)
+        end
+    end
+end
+
+--- Telescope Integration
+local pickers = require('telescope.pickers')
+local finders = require('telescope.finders')
+local actions = require('telescope.actions')
+local action_state = require('telescope.actions.state')
+
+--- Select a search engine using Telescope
+function M.select_search_engine()
+    local previous_search_engine = config.search_engine -- Save the previous search engine
+    pickers.new({}, {
+        prompt_title = "Select Search Engine",
+        finder = finders.new_table {
+            results = vim.tbl_keys(SearchEngineURLs),
+        },
+        attach_mappings = function(_, map)
+            map("i", "<CR>", function(prompt_bufnr)
+                local selection = action_state.get_selected_entry().value
+                actions.close(prompt_bufnr)
+                config.search_engine = selection -- Change the search engine
+                print("Searching with " .. selection)
+                -- Make the search with the selected search engine
+                vim.defer_fn(function()
+                    M.search()
+                end, 100)
+            end)
+            return true
+        end,
+    }):find()
+config.search_engine = previous_search_engine
+end
+
+--- Command to select search engine and only when the search engine is selected, search
+function M.select_search_engine_and_search()
+    M.select_search_engine()
 end
 
+--- Create user command for WebsearchOnEngine
+vim.api.nvim_create_user_command("WebsearchOnEngine", function()
+    M.select_search_engine_and_search()
+end, {})
+
+--- Create user command for Websearch History
+vim.api.nvim_create_user_command("WebsearchHistory", function()
+    pickers.new({}, {
+        prompt_title = "Search History",
+        finder = finders.new_table {
+            results = search_history,
+        },
+        attach_mappings = function(_, map)
+            map("i", "<CR>", function(prompt_bufnr)
+                local selection = action_state.get_selected_entry().value
+                actions.close(prompt_bufnr)
+                print("Searching for " .. selection)
+                -- Make the search with the selected history item
+                vim.defer_fn(function()
+                    local command = get_search_cmd(selection)
+                    if type(command) == "function" then
+                        command()
+                    else
+                        os.execute(command)
+                    end
+                end, 100)
+            end)
+            return true
+        end,
+    }):find()
+end, {})
+
+--- Create user command for delete search history
+vim.api.nvim_create_user_command("WebsearchHistoryDelete", function()
+    search_history = {}
+    save_history_to_file()
+    print("Search history deleted")
+end, {})
+
+--- Create user command for toggle w3m usage
+vim.api.nvim_create_user_command("WebsearchUseW3m", function()
```

Comment:
The proposal for w3m is good.
However, this approach would cause inconsistencies with the configuration settings.
For example, if false is set in the config, :WebsearchUseW3m would return true, but the file itself would remain as false.
I believe that this setting should be something users edit directly in the file.

Reply:
I definitely removed this one, I think it is better if the user modifies the configuration.